### PR TITLE
tracker: speed up create table by lower api

### LINKED ISSUE #1696: Speed up create table in scheamTracker
## Feature Request

**Is your feature request related to a problem? Please describe:**

Now DM embed a mockTiDB to record upstream schema info, every time we rollback a table from checkpoint, we will call `tracker.CreateTableIfNotExist`, but the func is too slow. create a table cost more than 2 second.
```
	ctx := context.Background()
	_ = tracker.Exec(ctx, "testdb", `
		create table foo(
			a int primary key auto_increment,
			b int as (c+1) not null,
			c int comment 'some cmt',
			d text,
			key dk(d(255))
		);
	`)

	// Save the table info
	ti1, _ := tracker.GetTable("testdb", "foo")
	ti1 = ti1.Clone()

	b.ResetTimer()
	for n := 0; n < b.N; n++ {
		tracker.CreateTableIfNotExists("testdb", fmt.Sprintf("foo-%d", n), ti1)
	}
```
go test -bench=. -run=none
```
goos: linux
goarch: amd64
pkg: github.com/pingcap/dm/pkg/schema
cpu: Intel(R) Xeon(R) CPU E5-2630 v4 @ 2.20GHz
BenchmarkTracker-40            1        2002442631 ns/op
PASS
ok      github.com/pingcap/dm/pkg/schema        2.313s
```
**Describe the feature you'd like:**
Speed up create table in scheamTracker

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
close #1696 

### What is changed and how it works?
use CreateTableWithInfo API provided by TiDB

### Check List 

Tests 

 - Unit test

Now create 100 tables cost 13 seconds

### CODE REVIEW

**Comment by lance6716** [comment]
@kennytm PTAL, is this DDL API suitable?

File: pkg/schema/tracker.go [hunk 1]
```diff
@@ -311,75 +306,8 @@ func (tr *Tracker) CreateTableIfNotExists(db, table string, ti *model.TableInfo)
...
-			}
-		}
-		time.Sleep(lease)
-	}
-	if err == nil {
-		// reaching here is basically a bug.
-		return errors.Errorf("Cannot create table %s.%s, the DDL job never returned", db, table)
-	}
-	return err
+	d := domain.GetDomain(tr.se).DDL()
```
Comment (Ehco1996) [inline pkg/schema/tracker.go:309]: do we need check d is nil ?

Comment (Ehco1996) [reply pkg/schema/tracker.go:309]: Rest LGTM

File: pkg/schema/tracker.go [hunk 2]
```diff
@@ -311,75 +306,8 @@ func (tr *Tracker) CreateTableIfNotExists(db, table string, ti *model.TableInfo)
...
+	ti = cloneTableInfo(ti)
+	ti.Name = tableName
+	return d.CreateTableWithInfo(tr.se, dbName, ti, ddl.OnExistError, false)
```
Comment (lance6716) [inline pkg/schema/tracker.go:312]: > ddl.OnExistError

please add a unit test about checking IfNotExists

File: pkg/schema/tracker.go (hunk 1 above)
Comment (lance6716) [inline pkg/schema/tracker.go:309]:
```suggestion
	d := tr.dom.DDL()
```

File: pkg/schema/tracker.go (hunk 2 above)
Comment (lance6716) [reply pkg/schema/tracker.go]: how did IfNotExists happen 😂

Comment (GMHDBJD) [reply pkg/schema/tracker.go]: should not happen，so I use error here

**Review by lichunzhu — COMMENTED** [review]
/lgtm

**Review by lance6716 — APPROVED** [review]

**Comment by lance6716** [comment]
```
[2021-05-18T10:56:26.546Z] FAIL: tracker_test.go:343: trackerSuite.TestCreateTableIfNotExists

[2021-05-18T10:56:26.546Z] 

[2021-05-18T10:56:26.546Z] tracker_test.go:419:

[2021-05-18T10:56:26.546Z]     c.Assert(duration.Seconds(), Less, float64(20))

[2021-05-18T10:56:26.546Z] ... compare_one float64 = 22.461430158

[2021-05-18T10:56:26.546Z] ... compare_two float64 = 20
```

**Comment by Ehco1996** [comment]
> /run-unit-tests

it-others test is not stable , i will fix it today , [jenkins](https://ci.pingcap.net/blue/organizations/jenkins/dm_ghpr_new_test/detail/dm_ghpr_new_test/221/pipeline)

**Pull request merged**